### PR TITLE
Add CNI chaining plugin for portmap

### PR DESCRIPTION
**Pull request by tgraf**
Enables users to add HostPort support by simply setting `cni-chaining-mode: portmap` in the ConfigMap.

This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7901)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

vagrant image download timed out

**Comment by tgraf** [comment]
test-me-please

vagrant image download timed out

**Comment by tgraf** [comment]
test-me-please

```
00:10:54.118  /opt/vagrant/embedded/gems/2.2.4/gems/vagrant-2.2.4/plugins/commands/up/middleware/store_box_metadata.rb:27:in `close': No space left on device @ fptr_finalize_flush - /home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/k8s-1.14-gopath/src/github.com/cilium/cilium/test/.vagrant/machines/k8s2-1.14/virtualbox/box_meta (Errno::ENOSPC)
```

**Comment by tgraf** [comment]
test-me-please

File: examples/kubernetes/templates/v1/cilium-cm.yaml [hunk 1]
```diff
@@ -155,3 +155,10 @@ data:
   # Enable legacy services (prior v1.5) to prevent from terminating existing
   # connections with services when upgrading Cilium from < v1.5 to v1.5.
   enable-legacy-services: "false"
+
+  # Enable chaining with another CNI plugin
```
Comment (aanm) [inline examples/kubernetes/templates/v1/cilium-cm.yaml:159]: This requires documentation changes as flannel will not work if `cni-chaining-mode` is not set with `flannel`

**Review by aanm — CHANGES_REQUESTED** [review]

File: examples/kubernetes/templates/v1/cilium-cm.yaml (hunk 1 above)
Comment (tgraf) [reply examples/kubernetes/templates/v1/cilium-cm.yaml:159]: Why does it not work?

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
test-docs-please

**Pull request merged**